modules/internal/ChapelIO.chpl

```diff
@@ -465,7 +465,7 @@ module ChapelIO {
         if st == QIO_AGGREGATE_FORMAT_JSON {
           start = new ioLiteral("{");
         } else if st == QIO_AGGREGATE_FORMAT_CHPL {
-          start = new ioLiteral("new " + typeToString(t) + "(");
+          start = new ioLiteral("new " + typeToString(t).c_str() + "(");
```

Comment:
I thought it also promoted the current support for casts to a user-facing feature?  But I guess your point is that one could write a module cast today without compiler modifications.

Offline, Kyle clarified for me that he doesn't support non-param casts to c_string because it would require something to deallocate the buffer for the string after the fact (this led me to ask whether we could disable all casts to c_string for uniformity).  In contrast, the .c_str() method points to the C string stored by the 'string' (which is a feature that will only work until we implement strings as something that doesn't make sense as a c_string...).